drivers/clock_control/clock_stm32_ll_common.c

```diff
@@ -232,6 +242,9 @@ static int stm32_clock_control_get_subsys_rate(const struct device *clock,
 	!defined (CONFIG_SOC_SERIES_STM32G0X)
 	uint32_t apb2_clock = get_bus_clock(ahb_clock, STM32_APB2_PRESCALER);
 #endif
+#if defined(CONFIG_SOC_SERIES_STM32WLX)
+	uint32_t apb3_clock = get_bus_clock(ahb_clock, STM32_AHB3_PRESCALER);
```

Comment:
Here you are assigning a different prescaler (STM32_AHB3_PRESCALER).

The apb3 prescaler is a good addition, but a few more steps are necessary to introduce it:
You need to introduce the apb3 flag in the device tree bindings before, then the symbol for STM32_APB3_PRESCALER, and additionally it would be fine if that prescaler is then set in the nucleo_board definitions.

I also had a short look on the adc part, will add a few comments after I had a more thorough look.

Reply:
Hi! From what I understand of the WL55/WLE5 clock architecture, the AHB3 (shared) prescaler output goes straight to both HCLK3 and PCLK3, so there's no APB3 prescaler in the chip. Guess I could `#define STM32_APB3_PRESCALER STM32_AHB3_PRESCALER`? Though that would sound more confusing than anything else to me.

Reply:
@ABOSTM Can you have a check ?

Reply:
Sorry, please ignore the comment above, I didn't check against the datasheet, and assumed it would have it's own prescaler.
In that case it may be preferable to call the variable ahb3_clock and add a fix to return this rate for STM32_CLOCK_BUS_AHB3 as well as STM32_CLOCK_BUS_APB3 in the switch case below?

Reply:
I confirm no APB3 dedicated prescaler,
variable apb3_clock, is good to me as this is the clock we want to compute.
Maybe add a comment to say HCLK3=APBCLK3 to avoid thinking there is a typo.

Reply:
Yeah that would work, but looking more into it I found another inconsistency: the current logic uses SystemCoreClock for STM32_CLOCK_BUS_AHB3, but that's incorrect for WL series. So I did this:

- renamed the variable to ahb3_clock (from AHB3 prescaler, so no confusion here)
- defined out the default STM32_CLOCK_BUS_AHB3 for the WL
- added STM32_CLOCK_BUS_AHB3 and STM32_CLOCK_BUS_APB3 to return ahb3_clock, with a comment

How does that look?

Reply:
Thanks, yes that is how I'd prefer it. Now only the issue about the SystemCoreClock raised by @ABOSTM should be left.